addons/medical_engine/functions/fnc_getHitpointArmor.sqf

```diff
@@ -0,0 +1,67 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Checks a unit's equipment to calculate the total armor on a hitpoint
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Hitpoint <STRING>
+ *
+ * Return Value:
+ * Total armor for the given hitpoint <ARRAY>
+ * 0: Armor <NUMBER>
+ * 1: ExplosionShielding <NUMBER>
+ *
+ * Example:
+ * [player, "HitChest"] call ace_medical_engine_fnc_getHitpointArmor
+ *
+ * Public: No
+ */
+params ["_unit", "_hitpoint"];
+
+// Helper function to check cache and/or read config
+private _fnc_getItemArmor = {
+    params ["_item", "_hitpoint", ["_isUniform", false]];
```

Comment:
I read your comment about you feeling checking if the item is a uniform or not is wasteful and I understand the intend.
However, you are missing the point. You're underestimating the power of the cache. By passing this as parameter, even if this parameter is basically never checked after the first execution, you are still investing extra time in this params command and the setting of a local variable.
This event handler will be executed many many times more than the caching event will take place. The cached isUniform check is on average lighter than this variable.

Reply:
hmm I suppose I follow - one extra config lookup per item is better than extra overhead on _every_ call. I struggle to believe that one extra param is worth it, but I'll defer to you since I'm by no means an expert.
Currently I'm only reading the config for the selected hitpoint, which means this will end up being called like a dozen times per-item; should I move back to caching all the hitpoints at once on the first call, to limit the overhead?

Reply:
That would involve adding a flag in the cache, to avoid that "have I cached this?" problem I mentioned, which is an extra `getVariable` on every execution so probably not, right?

Reply:
>I struggle to believe that one extra param is worth it, but I'll defer to you since I'm by no means an expert.

It's not a question of worth it. It's simply min-maxing what we can get out of this.

>Currently I'm only reading the config for the selected hitpoint, which means this will end up being called like a dozen times per-item; should I move back to caching all the hitpoints at once on the first call, to limit the overhead?

See? This is what I consider not worth it. This lookup is once per hitpoint per item. Completely neglible to the large number of repeated executions.